src/primitives/paraboloidsurface.jl

```diff
@@ -0,0 +1,115 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    ParaboloidSurface(point, radius, focallength)
+
+A paraboloid surface embedded in R³ and extending up to a distance `radius` from
+its focal axis, which is aligned along the z direction and passes through `point`
+(the apex of the paraboloid).
+
+    ParaboloidSurface(point, radius)
+
+This creates a paraboloid surface with focal length equal to 1.
+
+    ParaboloidSurface(point)
+
+This creates a paraboloid surface with focal length equal to 1 and a rim with unit
+radius.
+
+    ParaboloidSurface()
+
+Same as above, but here the vertex is at `Point(0, 0, 0)`.
+
+See also <https://en.wikipedia.org/wiki/Paraboloid>.
+"""
+struct ParaboloidSurface{T} <: Primitive{3, T}
+    point::Point{3,T}
+    radius::T
+    focallength::T
+end
+
+function ParaboloidSurface(point::Tuple, radius, focallength)
+    pt = Point(point)
+    ParaboloidSurface(pt, convert(eltype(pt), radius), convert(eltype(pt), focallength))
+end
+
+ParaboloidSurface(point::Point{3,T}, radius) where {T} = ParaboloidSurface(point, T(radius), one(T))
+
+ParaboloidSurface(point::Point{3,T}) where {T} = ParaboloidSurface(point, T(1), T(1))
+
+ParaboloidSurface{T}() where {T} = ParaboloidSurface(Point{3, T}(0, 0, 0), T(1), T(1))
+
+ParaboloidSurface() = ParaboloidSurface{Float64}()
```

Comment:
```suggestion
ParaboloidSurface() = ParaboloidSurface(Point(0, 0, 0), 1, 1)
```

Reply:
Mmm, with this change `ParaboloidSurface{Float32}()` does no longer work:

```julia
ParaboloidSurface: Error During Test at /home/tomasi/.julia/dev/Meshes/test/primitives.jl:1195
  Got exception outside of a @test
  MethodError: no method matching ParaboloidSurface{Float32}()
  
  Closest candidates are:
    (::Type{ParaboloidSurface{T}} where T)(::Any, ::Any, ::Any)
     @ Meshes ~/.julia/dev/Meshes/src/primitives/paraboloidsurface.jl:36
```

Reply:
Do we need this constructor with the explicit coordinate type? Our primitives don't have this special case. Whenever someone needs a special coordinate type, they can obtain it with the other constructors that rely on a specific point type.

Reply:
I removed the failing test and reverted your change.